integration_files/E2_Series/common_files/google_riscv_dv/lm_run.py

```diff
@@ -0,0 +1,1176 @@
+    cwd                   : Filesystem path to RISCV-DV repo
+  """
+  check_return_code = True
+  if argv.simulator == "ius":
+    # Incisive return non-zero return code even test passes
+    check_return_code = False
+    logging.debug("Disable return_code checking for %s" % argv.simulator)
+  # Mutually exclusive options between compile_only and sim_only
+  if argv.co and argv.so:
+    logging.error("argument -co is not allowed with argument -so")
+    return
+  if ((argv.co == 0) and (len(test_list) == 0)):
+    return
+  # Setup the compile and simulation command for the generator
+  compile_cmd = []
+  sim_cmd = ""
+  compile_cmd, sim_cmd = get_generator_cmd(argv.simulator, argv.simulator_yaml, argv.cov,
+                                           argv.exp, argv.debug);
+  # Compile the instruction generator
+  if not argv.so:
+    do_compile(compile_cmd, test_list, argv.core_setting_dir, cwd, argv.user_extension_dir,
+               argv.cmp_opts, output_dir, argv.debug, argv.lsf_cmd)
+  # Run the instruction generator
+  if not argv.co:
+    seed_gen = SeedGen(argv.start_seed, argv.seed, argv.seed_yaml)
+    do_simulate(sim_cmd, test_list, cwd, argv.sim_opts, seed_gen, argv.csr_yaml,
+                argv.isa, argv.end_signature_addr, argv.lsf_cmd, argv.gen_timeout, argv.log_suffix,
+                argv.batch_size, output_dir, argv.verbose, check_return_code, argv.debug)
+
+
+def gcc_compile(test_list, output_dir, isa, mabi, opts, gcc_user_extension_path, linker_path, debug_cmd):
+  """Use riscv gcc toolchain to compile the assembly program
+
+  Args:
+    test_list  				: List of assembly programs to be compiled
+    output_dir 				: Output directory of the ELF files
+    isa        				: ISA variant passed to GCC
+    mabi       				: MABI variant passed to GCC
+    gcc_user_extension_path	: Path to user extension directory containing defines files for tests etc
+    linker_path				: Path to linker file to be used in compilation
+    debug_cmd  				: Produce the debug cmd log without running
+  """
+########################################################
+##  generating assembly files for spike
+########################################################
+  # for test in test_list:
+  #   for i in range(0, test['iterations']):
+  #     if 'no_gcc' in test and test['no_gcc'] == 1:
+  #       continue
+  #     fin = open (("%s/asm_tests/%s_%d" %(output_dir, test['test'], i) + ".S"),"rt")
+  #     fout = open (("%s/asm_tests/%s_%d" %(output_dir, test['test'], i) + "_spike.S"),"wt")
+  #     do_write = True
+  #     for line in fin:
+  #       if do_write:
+  #         fout.write(line)
+  #       if "test_done:" in line:
+  #         do_write = False
+  #         continue
+  #       if (not do_write and "li gp, 1" in line):
+  #         do_write = True
+  #         fout.write(line)
+  #     fin.close()
+  #     fout.close()
+  # logging.info("Creating assembly files for spike")
+########################################################
```

Comment:
@saqibsaeed-lm What is the purpose of keeping this commented code block here?

Reply:
These were already commented and i have removed these lines